test/functional/tools/interactive_tool_juypter_notebook.xml

```diff
@@ -9,14 +9,9 @@
         </entry_point>
     </entry_points>
     <environment_variables>
-        <environment_variable name="HISTORY_ID" strip="True">
-            #if $jupyter_notebook:
-                ${__app__.security.encode_id($jupyter_notebook.history_id)}
-            #else:
-                ${__app__.security.encode_id($out_file1.history_id)}
-            #end if
-        </environment_variable>
-        <environment_variable name="REMOTE_HOST">TODO</environment_variable>
+        <environment_variable name="HISTORY_ID" strip="True">${__app__.security.encode_id($jupyter_notebook.history_id)}</environment_variable>
```

Comment:
We really need a way to inject this and something comparable to ``galaxy_infrastructure_url`` without ``__app__``. Willing to implement higher-level primitives for encoded history ID and inject ``galaxy_infrastructure_url`` with a per-job destination override option (https://github.com/galaxyproject/galaxy/pull/1688/commits/81b5ca564ac7cdd855cb74dad8e98abc83a8efdc). Or at least a warning in every sample that uses ``__app__`` that we will break backward compatibility in the near future

Reply:
Added in 725016230ef56f1e9f25d3018f3643dbfcc539cf and 34ec3bfd3a76a852dcbc9e637af395501e23a5ba